### PR TITLE
Add Viz function

### PR DESCRIPTION
**Pull request by Sai-prakash15**

### CODE REVIEW

File: frontend/src/Components/CustomizedInputBase.js [hunk 1]
```diff
@@ -29,14 +30,14 @@ export function CustomizedInputBase(props) {
     // console.log(input)
     props.apiCalled(true)
 
-    // const res = await axios.post('http://127.0.0.1:5000/api/v1/dashboard', {
-    //   input: input,
-    //   subgraph: subgraph,
-    // })
+    const res = await axios.post('http://127.0.0.1:5000/api/v1/dashboard', {
```
Comment (townie) [inline frontend/src/Components/CustomizedInputBase.js]: LGTM. Can we have this read from .env also?

**Review by townie — DISMISSED** [review]
shipit

File: frontend/src/Components/CustomizedInputBase.js (hunk 1 above)
Comment (Sai-prakash15) [reply frontend/src/Components/CustomizedInputBase.js]: YEP, one sec

Comment (Sai-prakash15) [reply frontend/src/Components/CustomizedInputBase.js]: DONE

**Review by townie — DISMISSED** [review]

**Comment by gosuto-inzasheru** [comment]
👀

frontend loads successfully for me in the vercel! just needs backend support now ?

File: frontend/src/Components/CustomizedInputBase.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ import {
   setData,
 } from "../redux/reducers/Counter/counter.actions"
 import { connect } from 'react-redux';
+import axios from 'axios';
+import { backend_url } from '../constants';
```
Comment (townie) [inline frontend/src/Components/CustomizedInputBase.js]: nice

File: frontend/src/Components/CustomizedInputBase.js [hunk 3]
```diff
@@ -25,20 +27,29 @@ export function CustomizedInputBase(props) {
...
+      res = await axios.post(`${backend_url}/api/v1/dashboard`, {
+      input: input,
+      subgraph: subgraph,
+    })
+    props.setData(res.data?.output);
+    props.apiCalled(false)
+  }
+    catch{
+      props.setData("");
+      props.setData([{"decimals":9,"id":"0xcf0c122c6b73ff809c693db761e7baebe62b6a2e","name":"FLOKI","symbol":"FLOKI","transferCount":274254},{"decimals":18,"id":"0x320623b8e4ff03373931769a31fc52a4e78b5d70","name":"Reserve Rights","symbol":"RSR","transferCount":"121409"},{"decimals":18,"id":"0xc5102fe9359fd9a28f877a67e36b0f050d81a3cc","name":"Hop","symbol":"HOP","transferCount":"78497"},{"decimals":18,"id":"0xa2cd3d43c775978a96bdbf12d733d5a1ed94fb18","name":"Chain","symbol":"XCN","transferCount":"70327"},{"decimals":9,"id":"0xa67e9f021b9d208f7e3365b2a155e3c55b27de71","name":"KleeKai","symbol":"KLEE","transferCount":"37061"}]);
```
Comment (townie) [inline frontend/src/Components/CustomizedInputBase.js]: I think we want this to display a "The query generation failed" or something to that effect vs displaying a random dataset

File: frontend/src/Components/Visulaizations/BarGraph.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import { Paper } from '@mui/material';
+import { Bar } from 'react-chartjs-2';
+// Chart.register(Bar);
+
+
+export default function BarVis(){
+    
+    const labels = ["Jan", "Feb", "mar", "apr", "may", "jun", "jul"];
```
Comment (townie) [inline frontend/src/Components/Visulaizations/BarGraph.js:8]: Tricky question, how are you going to pass this data long?

File: frontend/src/Components/CustomizedInputBase.js (hunk 3 above)
Comment (Sai-prakash15) [reply frontend/src/Components/CustomizedInputBase.js]: yep. I will change it. Just to see everything is working fine I have set that.

File: frontend/src/Components/Visulaizations/BarGraph.js (hunk 4 above)
Comment (Sai-prakash15) [reply frontend/src/Components/Visulaizations/BarGraph.js:8]: I will pass the data I get from backend as props to these components.

File: frontend/src/Components/CustomizedInputBase.js [hunk 5]
```diff
@@ -25,20 +27,29 @@ export function CustomizedInputBase(props) {
...
     // console.log(input)
     props.apiCalled(true)
-
-    // const res = await axios.post('http://127.0.0.1:5000/api/v1/dashboard', {
-    //   input: input,
-    //   subgraph: subgraph,
-    // })
+    
+    try{
+      res = await axios.post(`${backend_url}/api/v1/dashboard`, {
```
Comment (townie) [inline frontend/src/Components/CustomizedInputBase.js:37]: I am cherrypicking this

Comment (Sai-prakash15) [reply frontend/src/Components/CustomizedInputBase.js:37]: Let me push a new commit(commenting the viz part)

File: backend/backend/services/openAI/service.py [hunk 6]
```diff
@@ -30,6 +30,7 @@ def mapping_path(protocol):
 class OpenAIService:
     def __init__(self, use_prompt=0):
         openai.api_key = OPENAI_API_KEY
+        print(openai.api_key)
```
Comment (gosuto-inzasheru) [inline backend/backend/services/openAI/service.py]:
```suggestion
```

**Review by gosuto-inzasheru — APPROVED** [review]

**Pull request merged**